### DISCUSSION TITLE
errore "Cliente non trovato" durante creazione fattura

### DISCUSSION THREAD

**Discussion by Berubara83** [fattureincloud/api#178] [Q&A] [answered]
Ciao,

oggi durante la creazione di una fattura ho riscontrato il seguente errore
![IMG-20221130-WA0000](https://user-images.githubusercontent.com/112539870/204901081-d7a11ed9-7ff2-48de-ba3e-2d57268f2c5c.jpg)

Nella parte oscurata è presente il company_id recuperato con ListUserCompanies. 
A cosa si riferisce questo errore ?
Non ho mai riscontrato problemi nei caricamenti fatti da settembre fino al 29 Novembre 2022

Grazie

**Comment by emajo** [comment, marked as answer]
Ciao @Berubara83 l'errore riportato viene riscontrato quando si sta passando un id di cliente/fornitore inesistente, è una modifica fatta di recente.

Emanuele.

**Reply by valmoz** [reply]
Una precisazione: questa è una modifica dovuta ad un bugfix e legata all'ID dell'entity che viene inserita al momento della creazione dell'issued document.
L'ID dell'entity deve essere uno degli ID univoci generati dalle nostre API (insomma, gli ID che è possibile utilizzare per la getClient o la getSupplier). Se questo campo viene popolato con altro, avviene l'errore che avete riscontrato.

Mauro

**Reply by Berubara83** [reply]
Ciao,
grazie per le risposte.
 
Vorrei sapere se è obbligatorio specificare ID dell'entity o basta solo compilare le altre proprietà ?
Al momento non sto utilizzando CreateClient e GetClient ma i dati relativi al cliente risiedono sul mio gestionale.

Grazie mille

**Reply by valmoz** [reply]
L'id permette solo di collegare i clienti in anagrafica alla fattura, e serve soprattutto per clienti ricorrenti.
Per clienti sporadici, che non richiedono salvataggio in anagrafica, è sufficiente omettere il campo entity.id per evitare il tentativo di collegamento.

Può trovare maggiori informazioni qui: [Invoice Creation - The Customer](https://developers.fattureincloud.it/docs/guides/invoice-creation#1%EF%B8%8F%E2%83%A3-step-one-the-customer)

**Reply by Berubara83** [reply]
Grazie mille, funziona tutto!
